ts/Dashboards/Components/KPIComponent/KPIComponent.ts

```diff
@@ -447,12 +450,20 @@ class KPIComponent extends Component {
      *
      * @returns
      * The calculated average value.
-     *
-     * @internal
      */
-    private static calculateAverageValue(column: number[]): number {
-        const valueSum = KPIComponent.calculateSumValue(column);
-        return valueSum / column.length;
+    private static calculateAverageValue(column: DataTable.Column): number {
+        let sum = 0;
+        let count = 0;
+
+        for (let i = 0, iEnd = column.length; i < iEnd; ++i) {
+            const value = Number(column[i]);
+            if (!isNaN(value)) {
+                sum += value;
+                count++;
+            }
+        }
+
+        return count > 0 ? sum / count : 0;
```

Comment:
I think `NaN` indicates better that the calculation went wrong. When we declare all invalid values, we'll end up with the HC error either way. Should we consider returning `NaN` instead? I can see it may not be appropriate in terms of average calculation, but it seems to fit better in terms of the invalid data. Also, I'm referring to the `calculateMedianValue` when we return `NaN` in a similar case. 

<img width="723" alt="Screenshot 2025-02-07 at 10 12 55" src="https://github.com/user-attachments/assets/1778b8a6-3a84-46d7-9f53-eadc727a89d6" />

Reply:
Mathematically, the sum of an empty array is 0, not NaN. In the median case, the median of an empty array is NaN.